platform/local/tpm.go

```diff
@@ -3,26 +3,36 @@ package local
 import (
 	"fmt"
 	"os"
+	"path/filepath"
 
 	"github.com/coreos/pkg/capnslog"
 	"github.com/flatcar/mantle/system/exec"
 	"github.com/flatcar/mantle/util"
 )
 
 type SoftwareTPM struct {
-	process    *exec.ExecCmd
-	socketPath string
-	dir        string
+	process *exec.ExecCmd
+	dir     string
+	subDir  string
+	testDir string
 }
 
-func NewSwtpm(dir string) (*SoftwareTPM, error) {
-	swtpm := &SoftwareTPM{}
+func NewSwtpm(testDir string, tpmDir string) (*SoftwareTPM, error) {
+	dir := filepath.Join(testDir, tpmDir)
+	swtpm := &SoftwareTPM{dir: dir, subDir: tpmDir, testDir: testDir}
```

Comment:
this is super confusing :/
- testDir does not need to be stored in struct swtpm, it's not used after `NewSwtpm()`.
- get rid of the tpmDir parameter, and make the "tpm" dir internal to the class
- `SocketSubDirPath` -> `SocketRelativePath` (relative to the passed directory)
- dir -> something more expressive, cwd maybe?

Reply:
Yes, updated it now to use more descriptive names